### PR TITLE
Docs - Mention the Bazel version requirements in the README.md

### PR DESCRIPTION
**Pull request by NikeNano**
Update the readme with the Bazel version since it don't work with the latest versions. See this issue for more info: https://github.com/kubeflow/pipelines/issues/3942

### CODE REVIEW

File: components/gcp/container/component_sdk/python/kfp_component/google/dataflow/_client.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def __init__(self):
 
     def launch_template(self, project_id, gcs_path, location, 
         validate_only, launch_parameters):
-        return self._df.projects().templates().launch(
+        return self._df.projects().locations().templates().launch(
```
Comment (NikeNano) [inline components/gcp/container/component_sdk/python/kfp_component/google/dataflow/_client.py:24]:
```suggestion
         return self._df.projects().templates().launch(
```

File: components/gcp/container/component_sdk/python/kfp_component/google/dataflow/_client.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ def __init__(self):
 
     def launch_template(self, project_id, gcs_path, location, 
         validate_only, launch_parameters):
-        return self._df.projects().templates().launch(
+         return self._df.projects().templates().launch(
```
Comment (NikeNano) [inline components/gcp/container/component_sdk/python/kfp_component/google/dataflow/_client.py:24]:
```suggestion
        return self._df.projects().templates().launch(
```

**Comment by NikeNano** [comment]
Had some old commits that I had to clean up, thus all the commits.

**Comment by NikeNano** [comment]
friendly ping @rmgogogo

File: backend/README.md [hunk 3]
```diff
@@ -1,6 +1,11 @@
 This directory contains code for the components that comprise the Kubeflow
 Pipelines backend.
 
+
+## Bazel
+
+There are some [know issues](https://github.com/kubeflow/pipelines/issues/3942) with the latest version of Bazel and the kubeflow pipelines repo, therefore use version 0.24.1 or older.
```
Comment (Bobgy) [inline backend/README.md:7]: https://github.com/kubeflow/pipelines/search?q=bazel&unscoped_q=bazel

I think we are always using bazel 0.24.0 instead

**Comment by Bobgy** [comment]
other than that
/lgtm
Thanks for fixing the documentation!

File: backend/README.md (hunk 3 above)
Comment (NikeNano) [inline backend/README.md:7]:
```suggestion
There are some [know issues](https://github.com/kubeflow/pipelines/issues/3942) with the latest version of Bazel and the kubeflow pipelines repo, use version 0.24.0.  
```

Comment (NikeNano) [reply backend/README.md:7]: @Bobgy would this be a better way to state it? Maybe we dont need to mention the issues with later versions just that that 0.24.0 should be used?

Comment (Bobgy) [reply backend/README.md:7]: Yes, I think so. All developers and test infra is using this version, I think we should just tell people to use it.

**Comment by NikeNano** [comment]
Updated after feedback!

**Comment by Bobgy** [comment]
/lgtm
/approve
Thanks!

I think there are some outage with prow test infra, tests may keep failing, let's retest tomorrow to see if it's fixed

**Pull request merged**